tests/test_transformers.py

```diff
@@ -763,3 +764,118 @@ def test_binarize_works_in_gridsearch(self, train_iris_dataset):
         model = Model(grid)
         result = model.score_estimator(train_iris_dataset)
         assert isinstance(result, Result)
+
+
+class TestRareFeatureEncoder(TransformerBase):
+    def test_rare_feature_encoder_returns_correctly_series(self):
+        threshold = 2
+        fill_rare = "Rare"
+        rare = RareFeatureEncoder(threshold=threshold, fill_rare=fill_rare)
+
+        data = pd.DataFrame({"categorical": [1, "a", "a", 2, "b", 1]})
+        rare.fit(data)
+
+        new_data = pd.DataFrame({"categorical": [1, 1, 1, "a", "b", "b", 3]})
+        result = rare.transform(new_data)
+
+        test_data = pd.DataFrame({"categorical": [1, 1, 1, "a", "Rare", "Rare", 3]})
```

Comment:
would probably rename this to `expected` - in general, we want the pattern to be `assert result == expected`